lib/index.js

```diff
@@ -1,70 +1,58 @@
 'use strict';
 
-/**
- * Module dependencies.
- */
-
 var integration = require('@segment/analytics.js-integration');
+var useHttps = require('use-https');
 
 /**
- * Expose `Hindsight` integration.
+ * Expose `Hindsight`
+ *
  */
 
 var Hindsight = module.exports = integration('Hindsight')
-  .option('apiKey', '')
-  .tag('<script src="">');
+  .global('RB')
+  .option('pixel_code','')
+  .tag('http', '<script src="http://getrockerbox.com/assets/xyz.js">')
+  .tag('https', '<script src="https://getrockerbox.com/assets/xyz.js">');
 
 /**
- * Initialize.
- *
- * @api public
+ * Initialize
  */
 
 Hindsight.prototype.initialize = function() {
-  // put your initialization logic here
+  window.RB = {};
+  window.RB.disablePushState = true;
+  window.RB.queue = [];
+  window.RB.track = window.RB.track || function() {
+    window.RB.queue.push(Array.prototype.slice.call(arguments));
+  };
+  window.RB.source = this.options.pixel_code;
+
+  var protocol = useHttps() ? 'https' : 'http';
 
-  this.load(this.ready);
+  this.load(protocol, this.ready);
 };
 
 /**
- * Loaded?
- *
- * @api public
- * @return {boolean}
+ * Loaded
  */
 
 Hindsight.prototype.loaded = function() {
-  // what are required properties or functions that you need available on the `window`
-  // before the integration marks itself as ready?
-  return true;
+  return !!window.RB && !!window.RB.track;
 };
 
 /**
  * Page
- *
- * @api public
  */
 
-Hindsight.prototype.page = function(page) {
-  // send pageview data
-};
-
-/**
- * Identify
- *
- * @api public
- */
-
-Hindsight.prototype.identify = function(identify) {
-  // send user data
+Hindsight.prototype.page = function() {
+  window.RB.track('view');
 };
 
 /**
  * Track
- *
- * @api public
  */
 
 Hindsight.prototype.track = function(track) {
-  // send event data
+  window.RB.track(track.event(), track.properties());
```

Comment:
does this function take any free dictionary of key values? because customers can send whatever data format for track properties, including nested compound objects like arrays/objs

just wondering if we should do any sort of data massaging here